pytorch_lightning/trainer/connectors/training_trick_connector.py

```diff
@@ -24,24 +24,36 @@ def __init__(self, trainer):
 
     def on_trainer_init(
         self,
-        gradient_clip_val: float,
+        gradient_clip_val: Union[int, float],
         gradient_clip_algorithm: str,
         track_grad_norm: Union[int, float, str],
         accumulate_grad_batches: Union[int, Dict[int, int]],
         terminate_on_nan: bool,
     ):
-
-        self.trainer.terminate_on_nan = terminate_on_nan
+        if not isinstance(terminate_on_nan, bool):
+            raise MisconfigurationException(f"`terminate_on_nan` should be a bool, got {terminate_on_nan}.")
 
```

Comment:
some of these checks seem verbose because the argument is already typehinted as bool. why do we need to add specific runtime checks for those too?

also `TypeError` is the python standard exception for this situation

Reply:
not everyone takes care of the typehint. for eg if ArgumentParser is not configured correctly then sometimes passing `--terminate_on_bool True` converts `True -> 'True'`. The check just validates that everything is good at init.

updating to `TypeError`

Reply:
I think it's a choice that the framework needs to make.